src/math/integerhistogram1d.cpp

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#include "math/integerhistogram1d.h"
+#include "base/lineparser.h"
+#include "base/messenger.h"
+#include "base/sysfunc.h"
+#include "math/mathfunc.h"
+#include "templates/algorithms.h"
+#include <cassert>
+
+IntegerHistogram1D::IntegerHistogram1D()
+{
+    accumulatedData_.addErrors();
+
+    clear();
+}
+
+IntegerHistogram1D::IntegerHistogram1D(const IntegerHistogram1D &source) { (*this) = source; }
+
+// Clear Data
+void IntegerHistogram1D::clear()
+{
+    nBinned_ = 0;
+    nMissed_ = 0;
+    raw_.clear();
+    averages_.clear();
+}
+
+/*
+ * Data
+ */
+
+// Update accumulated data
+void IntegerHistogram1D::updateAccumulatedData()
+{
+    std::pair<Data1D, std::optional<int>> average = createDisplayData();
```

Comment:
For cleanliness this can be `auto`:
```suggestion
    auto average = createDisplayData();
```